### ISSUE TITLE
'revert' method breaks relations (after 2.8.1) 

### ISSUE THREAD

**Issue by emgolubev** [js-data/js-data#269]
It seems happens because of option 'replace' in inject method.

**Comment by davincho** [comment]
Could you create a plunkr for demonstration?

**Comment by davincho** [comment]
Ok the problem is that ``_this.previous`` only returns real attributes and no localKeys which are defined via a relation. As we inject only values of real attributes, the local key gets lost and so its relation to other objects.

**Comment by jmdobry** [comment]
Hmm, I'll have to dig into that test. In that instance, the user doesn't ever actually have any references to its comments, because it's the comments that each have a `userId` field.

**Comment by davincho** [comment]
Yes, the problem is that in https://github.com/js-data/js-data/blob/master/src/datastore/sync_methods/inject.js#L246 the attribute ``approvedBy`` will be deleted, as it is not part of the ``previousAttributes`` array of the instance.

**Comment by jmdobry** [comment]
I don't follow, the user and comments in your test don't use the `approvedBy` or `approvedByUser` fields.

**Comment by davincho** [comment]
Maybe I am wrong, but the hasMany relation ``comment`` in the ``User`` resource has as the foreignKey ``approvedBy``.
So the variable ``comment`` in my test is ```{ id: 11, userId: 10, content: 'test comment 11', approvedBy: 10 }`` but previousAttributes of that comment is ``{ id: 11, userId: 10, content: 'test comment 11' }`` - does that make sense?

**Comment by jmdobry** [comment]
I still don't understand where it's getting `approvedBy: 10` from, here is where comment 11 is created: https://github.com/js-data/js-data/blob/master/karma.start.js#L283 and https://github.com/js-data/js-data/blob/master/mocha.start.js#L270
